specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json

```diff
@@ -0,0 +1,2571 @@
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource name.",
+          "pattern": "^(?![0-9]+$)(?!-)[a-zA-Z0-9-]{2,49}[a-zA-Z0-9]$"
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource type."
+        },
+        "location": {
+          "type": "string",
+          "description": "The resource location."
+        },
+        "resourcegroup": {
+          "type": "string",
+          "description": "The resource group of the resource."
+        },
+        "subscriptionid": {
+          "type": "string",
+          "description": "The subscription id of the resource."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "The resource tags."
+        }
+      },
+      "x-ms-azure-resource": true,
+      "required": [
+        "location"
+      ]
+    },
+    "OperationListResult": {
+      "description": "Result of the request to list provisioning service operations. It contains a list of operations and a URL link to get the next set of results.",
+      "properties": {
+        "value": {
+          "description": "Provisioning service operations supported by the Microsoft.Devices resource provider.",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/Operation"
+          }
+        },
+        "nextLink": {
+          "readOnly": true,
+          "type": "string",
+          "description": "URL to get the next set of operation list results if there are any."
+        }
+      }
+    },
+    "Operation": {
+      "description": "Provisioning Service REST API operation.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Operation name: {provider}/{resource}/{read | write | action | delete}"
+        },
+        "display": {
+          "description": "The object that represents the operation.",
+          "properties": {
+            "provider": {
+              "readOnly": true,
+              "type": "string",
+              "description": "Service provider: Microsoft Devices."
+            },
+            "resource": {
+              "readOnly": true,
+              "type": "string",
+              "description": "Resource Type: ProvisioningServices."
+            },
+            "operation": {
+              "readOnly": true,
+              "type": "string",
+              "description": "Name of the operation."
+            }
+          }
+        }
+      }
+    },
+    "ErrorDetails": {
+      "description": "Error details.",
+      "type": "object",
+      "properties": {
+        "code": {
+          "description": "The error code.",
+          "format": "int32",
+          "type": "integer",
```

Comment:
These are highlighted as Breaking Changes. Please take a look, and get a breaking changes approval from Azure Breaking Changes Reviewers <azbreakchangereview@microsoft.com>.

Reply:
This is part of the pre-approved changes in private preview (per comment above). Let me know if we need to still go through another breaking change review.

Reply:
+1 on using common types to conform to RPC standard, instead of declaring your own error details

Error responses - use common types
There are definitions of error responses and details in commontypes types.json that you can use instead of declaring your own.
See e.g. ErrorResponse
[https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#L300](https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#L300)
And ErrorDetail
[https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#LL258C21-L258C21](https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#LL258C21-L258C21)

Reply:
If you're going to make breaking changes to the Error Response ANYWAY, wouldn't it make more sense to just be refactoring to using the standard definitions?? If they match your service behavior, that is.

Reply:
Turns out your service is nonstandard for returning int. ;-(